Unit tests for GXValidateCheckpointOperator

joshua-stauffer opened the pull request.

joshua-stauffer committed:
unit tests

tyler-hoffman commented on tests/unit/test_validate_checkpoint_operator.py:
```diff
@@ -66,3 +70,119 @@ def configure_checkpoint(context: AbstractDataContext) -> Checkpoint:
+        validate_checkpoint.execute(context={})
+
+        # assert
+        mock_gx.get_context.assert_not_called()
```
Seems like we should be asserting that we call configure_file_data_context and something is done with it.

tyler-hoffman reviewed (commented):
Looks good, except test_context_type_filesystem feels like it could use some other assertion that we actually do something with the callback.

Actually same deal with the configure_checkpoint functions. I think one test to check that it does its thing would be sufficient (i.e. don't bloat the existing cloud/ephemeral tests).

joshua-stauffer committed:
remove unused fixture

joshua-stauffer committed:
add test

joshua-stauffer commented:
thanks @tyler-hoffman , good suggestion. I added the test for `configure_file_data_context` in [b7d2a43](https://github.com/great-expectations/gx-airflow-provider/pull/8/commits/b7d2a43ef5b560036051630d5dca6919c6b88287). The previously existing test `test_batch_parameters` already covers the checkpoint.run case, so i updated the docstring to make that clear.

tyler-hoffman commented on tests/unit/test_validate_checkpoint_operator.py:
```diff
@@ -66,3 +70,143 @@ def configure_checkpoint(context: AbstractDataContext) -> Checkpoint:
+        def configure_file_data_context() -> FileDataContext:
+            return mock_file_data_context
+
+        def configure_checkpoint(context: AbstractDataContext) -> Checkpoint:
```
Probably easier to just pass a mock for this, and assert that the mock was passed the right context, but either way works.

joshua-stauffer replied:
yep, much clearer [479dd72](https://github.com/great-expectations/gx-airflow-provider/pull/8/commits/479dd726673d04b771697a2c339ed7d08cae40ae), thanks.

joshua-stauffer committed:
simplify tests to just mocks

joshua-stauffer committed:
Merge branch 'main' into f/core-727/checkpoint_operator_unit_tests
